src/Megadraft.js

```diff
@@ -56,6 +56,18 @@ class Megadraft extends Component {
     return false;
   }
 
+  handleReturn(event) {
+    if (event.shiftKey) {
+      const {editorState} = this.props;
+      const newState = RichUtils.insertSoftNewline(editorState);
+      if (newState) {
```

Comment:
Is this check needed? Looks like the function always returns a new state:
https://github.com/facebook/draft-js/blob/master/src/model/modifier/RichTextEditorUtil.js#L81